ftp.go

```diff
@@ -70,10 +72,31 @@ func Dial(addr string) (*ServerConn, error) {
 // It is generally followed by a call to Login() as most FTP commands require
 // an authenticated user.
 func DialTimeout(addr string, timeout time.Duration) (*ServerConn, error) {
+	return DialTimeoutWithConnection(addr, timeout, GetConnection)
+}
+
+func GetConnection(addr string, port int, timeout time.Duration) (net.Conn, error) {
+	// contains port suffix?
+	portSuffix := fmt.Sprintf(":%d", port)
+	if port > 0 && !strings.Contains(addr, portSuffix) {
+		addr = addr + portSuffix
+	}
```

Comment:
How is this supposed to work with IPv6? In which cases this the port number zero?